src/nl/hannahsten/texifyidea/inspections/latex/LatexCollapseCiteInspection.kt

```diff
@@ -123,18 +131,42 @@ open class LatexCollapseCiteInspection : TexifyInspectionBase() {
         }
 
         override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
-            val file = descriptor.psiElement.containingFile ?: return
-            val document = file.document() ?: return
             val sortedBundle = citeBundle.sortedBy { it.textOffset }
-            val offsetRange = sortedBundle.first().textOffset until sortedBundle.last().endOffset()
+            // The bundle can contain a gap when the cite commands in it surround a cite command
+            // that is not in the bundle, e.g., a cite command that has an optional parameter.
+            val bundleContainsGap = sortedBundle
+                .zipWithNext { a, b -> a.nextSibling != b }
+                .any()
 
+            // Create the content of the required parameter of the new cite command.
             val bundle = sortedBundle
                 .flatMap { it.requiredParameters }
                 .joinToString(",")
 
-            val first = citeBundle.first()
-            val star = if (first.hasStar()) "*" else ""
-            document.replaceString(offsetRange.toTextRange(), "${first.name}$star{$bundle}")
+            // Find the cite command that has to be replaced. When the bundle contains a gap, this is the command
+            // underneath the caret. When the bundle does not contain a gap this is the first command, as it doesn't matter
+            // whichever one we pick.
+            val targetCite = if (!bundleContainsGap) sortedBundle.firstOrNull() ?: return
+            else {
+                val editor = FileEditorManager.getInstance(project).selectedTextEditor
+                editor?.caretOffset()?.let {
+                    sortedBundle.first { cite -> cite.endOffset >= it }
+                }
+                    ?: sortedBundle.lastOrNull()
+                    ?: return
+            }
```

Comment:
this is a very interesting looking piece of code. I'd try to see if you can make it look a bit nicer. I'd suggest something like this (please check, just a quick thrown together piece):

```kotlin
            // Find the cite command that has to be replaced.
            // When the bundle contains a gap, this is the command underneath the caret.
            val targetCite = if (bundleContainsGap) {
                val editor = project.currentTextEditor()?.editor ?: return
                val currentCaret = editor.caretOffset()
                sortedBundle.first { it.endOffset >= currentCaret }
                /* sortedBundle.lastOrNull() can be omitted: this only happens when editor is 
                   null, which doesn't make sense (also early return because of that). */
            }
            // When the bundle does not contain a gap this is the first command, as it doesn't matter
            // whichever one we pick.
            else sortedBundle.firstOrNull() ?: return
```